torch/futures/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""
+The ``torch.futures`` package contains a ``Future`` type and corresponding
+utility functions.
+"""
+import torch
+
+
+class Future(torch._C.Future):
+    r"""
+    Wrapper around a ``torch._C.Future``.
+    """
```

Comment:
Shouldn't we have some details documentation here since this is a user facing API?

Reply:
Let me expand.